Changing to manual chart generation validation (#43)
ansible-community/awx-operator-helm: README.md

Chart-releaser is designed to use the `charts` directory as the source of truth for the current state of the chart.
If there are changes to that directory, the action generates a release.
Unlike many other helm charts, this one is generated on the fly by pulling in the awx-operator source code.

> As a result, any PR that changes the chart generation must run `make helm-chart-generate` and commit the chart changes as well.

Reviewer: ```suggestion
As a result, submitting a pull request that modifies the helm chart generation requires you to run `make helm-chart-generate` and commit the chart changes.
You should always create separate commits for your changes and the generated chart in your pull request.
```

Reviewer: Maybe it should go somewhere else or we should spell it out more but I think we should require separate commits for the output of the `make helm-chart-generate` command.

This also makes me wonder about the merge strategy. It might make sense to use merge commits for PRs that modify the contents of the charts directory.

Author: I think this is our main point of distance right now, I'm not sure I share the concern around having separate commits for chart generation since it should usually be clear how the starter changes will impact the chart. 

For example, there may be some redundancy in a PR which bumps the `appVersion` in starter's chart.yaml, and then has a second commit to generate the chart based on that appVersion

Of course, the ultimate policy is your call

Reviewer: I see your point but also feel pretty strongly about atomic commits. They have saved the day for me on a couple of occasions and I see them as best practice. Granted it's not always easy.

If we take this PR as an example, there is one commit that contains at least four distinct changes, the addition of the `committed-chart.yaml` workflow, the change to the `release` workflow, the doc updates in the README, as well as the generated chart.

If there were four separate commits for each of those things in this PR, it would aid with review. I know enough about the changes to see they are related but, being in a bit of a monolithic commit, it's much harder to see the logical separation between changes.

As well as that, each commit should have a meaningful message and be compact and focused. That helps tremendously with review by setting context. This also results in cleaner git history and makes it much easier to understand the rationale behind changes. When new folks join the project or there is a need to do any sort of retrospective investigation of the development over time, having cleanly separated commits makes it a whole lot easier to follow the "narrative".

Finally, atomic commits let you revert changes in a more precise way. For instance if we did decide to take a different approach and put the generated chart artifacts on a dedicated branch, it would be easier and cleaner to revert a single commit.

I could probably keep going about this but I don't want to labour the point or be too pedantic. I'm also not always right. I'm just trying to put this out there for consideration so we can get the best result.

Also to your statement that the ultimate policy is my call, I'm not trying to impose my view and respect what you feel is the right way to go with this. I see my role here as simply trying to help out and enable folks in the community to make this project work in a way that fits their needs.

Let me know what you think anyway. Thanks again for all the work you've put into this @miles-w-3

Author: I think your perspective makes sense. My experience has been at places which preach squashing commits as much as possible, but I see the merit of keeping independent changes within their own commits. I'm fine with the edited language you expressed in the suggestion, and thank you for the additional information

### Versioning

> The current CI setup will skip chart releases for PRs that don't make any changes to the generated chart.

Reviewer: ```suggestion
The current CI setup does not run chart releases for PRs that do not change the generated chart.
```

*Any* release that affects helm chart generation *must* increment the `version` field in `.helm/starter/Chart.yaml`, which is our source of truth for versioning in this repo. Follow the [semantic versioning](https://helm.sh/docs/topics/charts/#charts-and-versioning) guidelines outlined by the helm documentation.

> The `appVersion` field in the starter's `Chart.yaml` is the source of truth for the version of AWX Operator that will be pulled into the chart templates.

Reviewer: ```suggestion
The `appVersion` field in `.helm/starter/Chart.yaml` is the source of truth for the version of AWX Operator that is pulled into the chart templates.
```

> > Before version 3.0.0 of this Chart, the `version` field was tied to the AWX Operator version. Now, these are separate fields which can be incremented to reflect changes to the chart as well as the underlying app it installs.

Reviewer: ```suggestion
> Before version 3.0.0 of this Chart, the `version` field matched the AWX Operator version. The `version` and `appVersion` fields are independent and can be incremented separately to reflect changes to the chart or the underlying app it installs.
```
